app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_posts_count
+    author.increment!(:posts_counter)
+  end
```

Comment:
- Good job done so far, but according to the project requirements, I suggest you call this function after the data has been saved to your database. Please fix this 🙏 i.e `after_save :update_posts_count`